DOCUMENTATION.md

```diff
@@ -0,0 +1,296 @@
+
+Using the client in a browser is similar, except you don’t need to polyfill WebSocket (browsers have it natively) and you will access the library via the global `cdplogger` (if included via `<script>` tag) or via an ES module import (if using a bundler). 
+
+**Example (Browser with script includes):**
+
+```html
+<!-- Include scripts as described in Installation -->
+<script src="protobuf.min.js"></script>
+<script src="containerPb.js"></script>
+<script src="client.js"></script>
+
+<script>
+  // Once scripts are loaded, use the global cdplogger
+  const client = new cdplogger.Client("ws://localhost:17000");  // connect to logger
+  
+  // Example: fetch API version
+  client.requestApiVersion().then(version => {
+    console.log("Connected to CDP Logger, version:", version);
+  });
+  
+  // Example: fetch logged nodes and display in console
+  client.requestLoggedNodes().then(nodes => {
+    console.log("Logged values available:", nodes.map(n => n.name));
+  });
+  
+  // You could then use this data to, for instance, populate a UI or chart.
+</script>
+```
+
+As shown, the usage is very much the same as in Node. The `cdplogger.Client` class is the main entry point. In a browser context, you might integrate this with UI components or state management. For example, in a Vue.js app, you might call these client methods in a Vuex store or inside component lifecycle hooks to retrieve data and then render charts (as demonstrated in the included Vue example). The provided `model.js` (in this repository’s examples) shows a more involved integration: it uses the client to load historical data for multiple sensors and plot them using Chart.js, and to fetch events (like alarms) to display in a list. You can refer to that for inspiration on how to bind the data to a frontend.
+
+A few browser-specific notes:
+
+- **Security:** If your CDP Logger is served over HTTPS (or you are accessing it from an HTTPS page), use `wss://` for the WebSocket URL to avoid mixed content issues. Ensure the CDP Logger is configured with TLS if needed. For local testing with `http://localhost` you can use `ws://` without issues.
+
+- **CORS/WebSocket Policy:** WebSocket connections are not subject to the same-origin policy in the way XHR/Fetch are, but some environments might still require the server to accept the connection. The CDP Logger’s WebSocket server should accept connections from any origin by default. If you encounter issues connecting from a web page, check if any firewall or network configuration is blocking the websocket port.
+
+- **Performance:** Retrieving a lot of data points or events in one go can be heavy for the browser. If you plan to visualize large data sets, consider using pagination (e.g., request events 100 at a time using `offset`) or downsampling data points via the `numberOfPoints` parameter. The library itself streams the data efficiently, but rendering thousands of points in the DOM can be slow, so plan accordingly.
```

Comment:
The quoted `numberOfPoints` does not seem to be a valid param name - make sure quoted stuff is exact.
![image](https://github.com/user-attachments/assets/b8579099-a1cf-4670-897f-d3e420275bc3)